### PR TITLE
fix CI issues

### LINKED ISSUE #389: Limit concurrency to 1 in the CI
We should limit the concurrency to 1 in the CI, following -> https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#concurrency

This will save some potential ordering problems

### PR DESCRIPTION
**Pull request by stefanotorresi**
fixes #389 and #354 
it also make the ansible linting and python checking in parallel with the go testing, which should speed up things considerably.

### CODE REVIEW

**Review by fabriziosestito — APPROVED** [review]
Thanks for this!

File: requirements.dev.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ansible==4.6.0
```
Comment (arbulu89) [inline requirements.dev.txt:1]: Good idea!
Most probably, we could even have a `prod` requirements file, that is used in runner container creation.

**Review by arbulu89 — APPROVED** [review]
Great!

File: requirements.dev.txt (hunk 1 above)
Comment (stefanotorresi) [reply requirements.dev.txt:1]: yeah, I thought about this! Probably in another PR though.

**Pull request merged**